packages/service/src/utils/doc-persistence.ts

```diff
@@ -0,0 +1,55 @@
+import db from "./db";
+import { Prisma, document } from "@prisma/client";
+
+export const insertUpdate = async (
+  documentId: string,
+  documentUpdate: Uint8Array
+): Promise<any> => {
+  return await db.document_updates_queue.create({
+    data: {
+      document_id: documentId,
+      document_update: Buffer.from(documentUpdate),
+    },
+  });
+};
+
+export const createDocument = async (
+  initState: Uint8Array,
+  initValue: string,
+  initWeb_rtc_key: string
+): Promise<Prisma.Prisma__documentClient<document>> => {
+  return await db.document.create({
+    data: {
+      value: initValue,
+      origin: Buffer.from(initState),
+      web_rtc_key: initWeb_rtc_key,
+      latest_update_time: new Date(Date.now()),
+    },
+  });
+};
+
+export const loadDocument = async (documentId: string): Promise<any> => {
+  // TODO:
+  // find document by id in docs table
+  return db.document.findFirst({
+    where: {
+      id: documentId,
+      document_updates_queue: {
+        every: { document_id: documentId },
+      },
+    },
+    include: {
+      document_updates_queue: true,
```

Comment:
the join here needs to be on a range I think where `document_updates_queue.created_at > document.latest_update_time`

Reply:
Right, good catch